src/NumericalAlgorithms/SphericalHarmonics/YlmSpherepack.cpp

```diff
@@ -609,6 +612,14 @@ void YlmSpherepack::interpolate_from_coefs(
     const InterpolationInfo<T>& interpolation_info,
     const size_t spectral_stride, const size_t spectral_offset) const {
   if (storage_.work_interp_alpha.empty()) {
+    ASSERT(m_max_ == interpolation_info.m_max(),
+           "Different m_max for InterpolationInfo ("
+               << interpolation_info.m_max() << ") and YlmSpherepack instance ("
+               << m_max_ << ")");
```

Comment:
Should these be ERRORs instead of ASSERTs?   (I can imagine cases where l_max and m_max change dynamically during a run, and an ASSERT won't catch this error in release mode.)

Reply:
Since the cost is virtually zero that makes sense.